shared/chat/conversation/list/index.desktop.js

```diff
@@ -264,32 +264,44 @@ class PopupEnabledList extends BaseList {
 
   // How this works is kinda crappy. We have to plumb through this key => message helper and all this DOM stuff just to support this
   _onEditLastMessage = () => {
-    const entry: any = this.props.messageKeys.findLastEntry(k => {
-      const m = this.props.getMessageFromMessageKey(k)
-      return !!(m && m.type === 'Text' && m.author === this.props.you)
+    let tuple: ?[number, Constants.MessageKey, Constants.TextMessage]
+    this.props.messageKeys.findLastEntry((v, k) => {
+      const m = this.props.getMessageFromMessageKey(v)
+      if (m && m.type === 'Text' && m.author === this.props.you) {
+        tuple = [k, v, m]
+        return true
+      }
+      return false
     })
 
-    if (entry) {
-      const idx: number = entry[0]
-      const messageKey: Constants.MessageKey = entry[1]
-      // $ForceType
-      const message: Constants.TextMessage = this.props.getMessageFromMessageKey(messageKey)
-
-      this._keepIdxVisible = idx
-      this.setState({listRerender: this.state.listRerender + 1})
-
-      const listNode = ReactDOM.findDOMNode(this._list)
-      if (listNode) {
-        // $FlowIssue
-        const messageNodes = listNode.querySelectorAll(`[data-message-key="${messageKey}"]`)
-        if (messageNodes) {
-          const messageNode = messageNodes[0]
-          if (messageNode) {
-            this._showEditor(message, this._domNodeToRect(messageNode))
-          }
-        }
-      }
+    if (!tuple) {
+      return
+    }
+
+    const [idx, messageKey, message] = tuple
+    if (!Constants.textMessageEditable(message)) {
+      return
+    }
+
+    this._keepIdxVisible = idx
+    this.setState({listRerender: this.state.listRerender + 1})
```

Comment:
to be non-racy this should be
```
this.setState(prevState => {listRerender: prevState.listRerender + 1})
```

Reply:
Done. Looks like there's another instance in this file (and possibly other related files) but I'll do those separately